ext/standard/math.c

```diff
@@ -1096,9 +1096,8 @@ PHP_FUNCTION(number_format)
 {
 	double num;
 	zend_long dec = 0;
-	char *thousand_sep = NULL, *dec_point = NULL;
-	char thousand_sep_chr = ',', dec_point_chr = '.';
-	size_t thousand_sep_len = 0, dec_point_len = 0;
+	char *thousand_sep = ",", *dec_point = NULL;
```

Comment:
Why not 
```suggestion
	char *thousand_sep = ",", *dec_point = ".";
```
?

Reply:
I was hitting an issue when ``null`` is passed as the 3rd argument in which the decimal point char* gets reset to NULL. If you have a better idea how to circumvent this I'd like to hear it because I didn't give it much thought.

Reply:
I see. Shouldn't thousand_sep get the same treatment though? Also, just use `dec_point = "."` below, no need to take a reference to a char.

Reply:
Right, I only caught the former because of a bug test case and then ignored the fact that the thousand seperator could also be passed null.
Added a test case for those cases.